solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java

```diff
@@ -754,13 +761,12 @@ protected void handle(List<PrometheusMetric> results, JsonNode metricsNode) thro
           if (!INVALID_NUMBER.equals(value)) {
             results.add(metric.createPrometheusMetric(value, "[node aggregated]"));
           } else {
-            missingCoreMetrics.add(metric);
+            resultContext.missingCoreMetrics.add(metric);
           }
         }
       } else {
         log.warn(
             "Cannot find the solr.node metrics, going to fall back to getting metrics from all cores");
-        missingCoreMetrics.addAll(Arrays.asList(CoreMetric.values()));
       }
```

Comment:
this is removed here because absence of `missingCoreMetrics` is interpreted by the downstream metrics caller as "the aggregate one was either not called, or didn't receive any metrics", so the effect is the same?

Reply:
Yes. I want to achieve looser coupling, that any metrics caller can set missingCoreMetrics to explicitly identify that it has checked the core metrics, and found all the missing core metrics and put them in `missingCoreMetrics` (could be empty, which means an upstream caller has already performed a check, and found no missing core metrics at all)

And in the `CoresMetricsApiCaller`, it can then check in the `ResultContext` and see if any previous metrics caller has already checked core metrics. If not (null), then fetch all core metrics, if set (non null), fetch accordingly to those missing ones.

This enables `CoresMetricsApiCaller` to make less assumption on the dependency. For example, if we remove `AggregateMetricsApiCaller` from the calling chain, `CoresMetricsApiCaller` should still work as is 😊  (and also unit test case is cleaner ✨ )